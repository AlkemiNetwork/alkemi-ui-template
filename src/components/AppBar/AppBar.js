import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import MuiAppBar from "@material-ui/core/AppBar";
import { withStyles } from "@material-ui/core/styles";

import AppBarToolbar from "./Toolbar";

const styles = theme => ({
    appBar: {
        position: "fixed",
        top: 0,
        zIndex: theme.zIndex.drawer + 1,
        transition: theme.transitions.create(["width", "margin"], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen
        })
    },
    appBarShifted: {
        marginLeft: 240,
        width: `calc(100% - ${240}px)`,
        transition: theme.transitions.create(["width", "margin"], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen
        }),
        [theme.breakpoints.down("sm")]: {
            width: "100%"
        }
    }
});

const AppBar = ({ classes, shifted }) => {
    return (
        <MuiAppBar
            position="absolute"
            className={classNames(
                classes.appBar,
                shifted && classes.appBarShifted
            )}
        >
            <AppBarToolbar shifted={shifted} />
        </MuiAppBar>
    );
};

AppBar.propTypes = {
    classes: PropTypes.object.isRequired,
    shifted: PropTypes.bool
};

export default withStyles(styles)(AppBar);
