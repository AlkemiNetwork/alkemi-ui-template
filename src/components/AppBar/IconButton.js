import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";

const styles = theme => ({
    menuButton: {
        marginLeft: 12,
        marginRight: 36
    },
    hide: {
        [theme.breakpoints.up("sm")]: {
            display: "none"
        }
    }
});

const AppbarButton = ({ classes, onClick, shifted }) => (
    <IconButton
        color="inherit"
        aria-label="open drawer"
        onClick={onClick}
        className={classNames(classes.menuButton, shifted && classes.hide)}
    >
        <MenuIcon />
    </IconButton>
);

AppbarButton.propTypes = {
    onClick: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    shifted: PropTypes.bool
};

export default withStyles(styles)(AppbarButton);
