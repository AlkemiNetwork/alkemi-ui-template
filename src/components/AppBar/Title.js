import React from "react";
import PropTypes from "prop-types";

import ButtonBase from "@material-ui/core/ButtonBase";
import Typography from "@material-ui/core/Typography";

const AppBarTitle = ({ onClick }) => (
    <ButtonBase onClick={onClick}>
        <Typography variant="title" color="inherit" noWrap>
            ConsenSys React Boilerplate
        </Typography>
    </ButtonBase>
);

AppBarTitle.propTypes = {
    onClick: PropTypes.func.isRequired
};

export default AppBarTitle;
