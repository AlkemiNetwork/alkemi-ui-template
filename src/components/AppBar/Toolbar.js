import React from "react";
import PropTypes from "prop-types";

import Toolbar from "@material-ui/core/Toolbar";
import withWidth, { isWidthUp } from "@material-ui/core/withWidth";

import AppBarTitle from "../../containers/AppBar/Title";
import AppBarIconButton from "../../containers/AppBar/IconButton";

const AppBarToolbar = ({ width, shifted }) => (
    <Toolbar disableGutters={!shifted || isWidthUp("md", width)}>
        <AppBarIconButton />
        <AppBarTitle />
    </Toolbar>
);

AppBarToolbar.propTypes = {
    width: PropTypes.string.isRequired,
    shifted: PropTypes.bool
};

export default withWidth()(AppBarToolbar);
