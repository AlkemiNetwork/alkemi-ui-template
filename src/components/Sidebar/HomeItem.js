import React from "react";
import PropTypes from "prop-types";

import HomeIcon from "@material-ui/icons/Home";

import ListItem1 from "../UI/ListItem/1";

const SidebarHomeItem = ({ onClick, selected }) => (
    <ListItem1
        icon={<HomeIcon />}
        text="Home"
        onClick={onClick}
        selected={selected}
    />
);

SidebarHomeItem.propTypes = {
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired
};

export default SidebarHomeItem;
