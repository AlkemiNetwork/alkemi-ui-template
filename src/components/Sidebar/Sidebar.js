import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { compose } from "redux";

import Drawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import withWidth, { isWidthUp } from "@material-ui/core/withWidth";
import { withStyles } from "@material-ui/core/styles";

import SidebarHomeItem from "../../containers/Sidebar/HomeItem";
import SidebarHeader from "../../containers/Sidebar/Header";

const styles = theme => ({
    drawer: {
        height: "100vh",
        position: "fixed"
    },
    drawerPaper: {
        position: "relative",
        height: "100%",
        width: 240,
        transition: theme.transitions.create("width", {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen
        })
    },
    drawerPaperClose: {
        width: 0,
        overflowX: "hidden",
        transition: theme.transitions.create("width", {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.leavingScreen
        }),
        [theme.breakpoints.up("sm")]: {
            width: 60
        }
    },
    drawerInner: {
        // Make the items inside not wrap when transitioning:
        width: 240
    }
});

const Sidebar = ({ classes, width, variant, open, onClose }) => {
    const isLargeScreen = isWidthUp("sm", width);
    return (
        <Drawer
            variant={isLargeScreen ? "permanent" : undefined}
            classes={
                isLargeScreen
                    ? {
                          docked: classes.drawer,
                          paper: classNames(
                              classes.drawerPaper,
                              !open && classes.drawerPaperClose
                          )
                      }
                    : {}
            }
            open={open}
            onClose={onClose}
        >
            <div className={classes.drawerInner}>
                <SidebarHeader />
                <List className={classes.list}>
                    <SidebarHomeItem />
                </List>
            </div>
        </Drawer>
    );
};

Sidebar.propTypes = {
    classes: PropTypes.object.isRequired,
    width: PropTypes.oneOf(["xs", "sm", "md", "lg", "xl"]),
    variant: PropTypes.string,
    open: PropTypes.bool,
    onClose: PropTypes.func
};

export default compose(withWidth(), withStyles(styles))(Sidebar);
