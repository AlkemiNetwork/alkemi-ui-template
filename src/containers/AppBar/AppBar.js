import { connect } from "react-redux";

import AppBar from "../../components/AppBar/AppBar";

const mapStateToProps = state => ({
    shifted: state.ui.sidebar.open
});

export default connect(mapStateToProps)(AppBar);
