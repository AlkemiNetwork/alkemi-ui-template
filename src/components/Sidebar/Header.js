import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import IconButton from "@material-ui/core/IconButton";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";

const styles = theme => ({
    drawerHeader: {
        display: "flex",
        alignItems: "center",
        justifyContent: "flex-end",
        padding: "0 " + theme.spacing.unit + "px",
        ...theme.mixins.toolbar
    }
});

const SidebarHeader = ({ classes, onClick }) => {
    return (
        <div className={classes.drawerHeader}>
            <IconButton onClick={onClick}>
                <ChevronLeftIcon />
            </IconButton>
        </div>
    );
};

SidebarHeader.propTypes = {
    classes: PropTypes.object.isRequired,
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired
};

export default withStyles(styles)(SidebarHeader);
