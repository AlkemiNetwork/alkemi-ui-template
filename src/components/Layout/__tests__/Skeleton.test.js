import React from "react";
import toJson from "enzyme-to-json";

import LayoutSkeleton from "../Skeleton";
import LayoutContent from "../../../containers/Layout/Content";
import AppBar from "../../../containers/AppBar/AppBar";
import Sidebar from "../../../containers/Sidebar/Sidebar";
import createShallow from "../../../tests-utils/createShallow";

describe("<LayoutSkeleton />", () => {
    let shallow;
    let wrapper;

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });

    beforeEach(() => {
        wrapper = shallow(<LayoutSkeleton />);
    });

    it("renders 1 <LayoutContent /> components", () => {
        expect(wrapper.find(LayoutContent).length).toEqual(1);
    });

    it("renders 1 <AppBar /> components", () => {
        expect(wrapper.find(AppBar).length).toEqual(1);
    });

    it("renders 1 <Sidebar /> components", () => {
        expect(wrapper.find(Sidebar).length).toEqual(1);
    });

    it("matches snapshot", () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
