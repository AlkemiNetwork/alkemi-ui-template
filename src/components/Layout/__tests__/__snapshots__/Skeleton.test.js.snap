// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutSkeleton /> matches snapshot 1`] = `
<div
  className="LayoutSkeleton-frame-1"
>
  <Connect(WithStyles(AppBar)) />
  <Connect(WithTheme(WithWidth(WithStyles(Sidebar)))) />
  <div
    className="LayoutSkeleton-content-2"
  >
    <LayoutContent />
  </div>
</div>
`;
